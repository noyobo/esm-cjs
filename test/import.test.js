const transform = require('../lib/transform');

describe('transform import', () => {
  it('should star import', () => {
    expect(transform(`import * as foo from 'foo';`)).toMatchSnapshot();
  });

  it('should default import', () => {
    expect(transform(`import foo from 'foo';`)).toMatchSnapshot();
  });

  it('should import members', () => {
    expect(transform(`import { a, b, c } from 'foo';`)).toMatchSnapshot();
  });

  it('should import members alias', () => {
    expect(transform(`import { a as d, b as e, c } from 'foo';`)).toMatchSnapshot();
  });

  it('should import members and default', () => {
    expect(transform(`import d, { a, b, c } from 'foo';`)).toMatchSnapshot();
  });
});
