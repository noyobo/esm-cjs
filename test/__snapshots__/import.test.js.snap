// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform import should default import 1`] = `"const foo = require('foo');"`;

exports[`transform import should import members 1`] = `
"const _foo = require('foo');

const a = _foo.a;
const b = _foo.b;
const c = _foo.c;"
`;

exports[`transform import should import members alias 1`] = `
"const _foo = require('foo');

const d = _foo.a;
const e = _foo.b;
const c = _foo.c;"
`;

exports[`transform import should import members and default 1`] = `
"const d = require('foo');

const a = d.a;
const b = d.b;
const c = d.c;"
`;

exports[`transform import should star import 1`] = `"const foo = require('foo');"`;
